### PR TITLE
PHP 8 + PHPUnit 9.4

### PR DESCRIPTION
**Pull request by rodber**
Hi,

This one adds support for PHP 8 and PHPUnit 9.4. I had to drop support for 5.4.

Tested locally with 7.4 + 8.0. Travis passing.

### CODE REVIEW

**Comment by Xantios** [comment]
Can someone please merge this so I can pull this down from packagist ?:-) Thanks !

**Comment by peter279k** [comment]
pin @ralouphie.

**Review by peter279k — APPROVED** [review]

**Comment by dmyers** [comment]
I'm wondering if this is even maintained anymore. The last commit to `develop` was from March 2019.

**Comment by Xantios** [comment]
> I'm wondering if this is even maintained anymore. The last commit to `develop` was from March 2019.

I Assumed the same. so i forked this repo and made sure it works in PHP 8.
https://github.com/xantios/mimey

I just added a custom VCS entry to my composer file, but if there is any interest I can push it to packagist

**Comment by peter279k** [comment]
Pin @ralouphie.

**Review by pvsaintpe — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
 php:
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7.0
-  - 7.1
-  - 7.2
   - 7.3
+  - 7.4
+  - nightly
```
Comment (pvsaintpe) [inline .travis.yml:6]: Please, add 8.0

**Review by pvsaintpe — CHANGES_REQUESTED** [review]

**Comment by pvsaintpe** [comment]
@ralouphie, @rodolfoberrios 

We really need support in our project:
 - php: ^8.0
 - phpunit/phpunit: ^9.5
 - php-coverals: ^2.0

When can we expect a release with these changes? I really don't want to make forks and other crutches ..

Thank you in advance.

**Comment by rodber** [comment]
@pvsaintpe 

I just contributed a few lines, I'm not the maintainer of this project. It seems that the person in charge isn't around, and likely it won't appear if you just keep mention this issue. Please don't push, it doesn't bring anything constructive imo. It will sound harsh, but if you really need this so bad then consider to fork it and maintain it. Otherwise you will be fall in the same situation with other maintainers not matching your release schedule requirements.

**Review by pvsaintpe — APPROVED** [review]

**Comment by vedro-compota** [comment]
@ralouphie this needed

**Comment by vedro-compota** [comment]
Guys we need some fork or does anyone have contact of the Author?
@Xantios yes, packagist for this pack is needed too, because of recurs. dependencies.

**Comment by rodber** [comment]
I'm done, sorry.

**Comment by Xantios** [comment]
> Guys we need some fork or does anyone have contact of the Author?
> @Xantios yes, packagist for this pack is needed too, because of recurs. dependencies.

You can add a VCS entry to your composer file and pull my fork in.
If there are more people wanting to maintain it, feel free to PR on that fork I will setup some composer stuff around it later today

**Comment by Xantios** [comment]
@rodolfoberrios No worries! 👍 Thanks for figuring this out!
@vedro-compota Check out my fork on https://github.com/xantios/mimey

For all, if you want to PR something, feel free to PR my fork 👍

**Comment by peter279k** [comment]
> @rodolfoberrios No worries!  Thanks for figuring this out!
> @vedro-compota Check out my fork on https://github.com/xantios/mimey
> 
> For all, if you want to PR something, feel free to PR my fork

@Xantios, thanks a lot!

**Comment by peter279k** [comment]
@Xantios, BTW, could you create the issue on your forked repository?

**Comment by Xantios** [comment]
> @Xantios, BTW, could you create the issue on your forked repository?

The issue (PHP 8 / PHPUnit 9.4 support) is already merged in on the fork :-)

**Comment by peter279k** [comment]
> > @Xantios, BTW, could you create the issue on your forked repository?
> 
> The issue (PHP 8 / PHPUnit 9.4 support) is already merged in on the fork :-)

Sorry. I mean could you enable issue setting on your forked repository?

You can enable the issue like following picture:

image

**Comment by Xantios** [comment]
Done! @peter279k

**Comment by Radon8472** [comment]
You should remove

```php
protected function setUp(): void
```
From you PR, this breaks compatibility to older php versions

**Pull request closed without merging**